src/org/opendatakit/briefcase/ui/ScrollingStatusListDialog.java

```diff
@@ -128,8 +130,12 @@ private ScrollingStatusListDialog(Frame frame, String labelText, String title, I
   public void onEvent(FormStatusEvent event) {
     // Since there can be multiple FormStatusEvent's published concurrently,
     // we have to check if the event is meant for this dialog instance.
-    if (event.getStatus().getFormDefinition().equals(form)) {
-      editorArea.setText(event.getStatus().getStatusHistory());
+    if (isShowing() && event.getStatus().getFormDefinition().equals(form)) {
+      try {
+        editorArea.read(new StringReader(event.getStatus().getStatusHistory()), null);
```

Comment:
Thanks for the clarifications -- I didn't realize that the issue was the total memory usage, I thought it was only the repeated allocation and deallocation of ever larger objects. I would have expected the buffer to be able to hold several MBs without trouble on a modern computer!